### PR TITLE
Vietnamese marriage outcomes

### PR DESCRIPTION
**Pull request by ikennaokpala**
[Trello card](https://trello.com/c/lxtegDlR/234-vietnam-marriage-tool-update-content-change-request)

## Description

This PR updates the content and links for Vietnamese marraiges:
- Add content for same sex marrigages where partner is Vietnamese National
- Update Content and links for Opposite and same sex marriages in Vietnam

## Factcheck Same sex marriage for Vietnamese Nationals

[Preview link](https://smart-answers-pr-2721.herokuapp.com/marriage-abroad/y/vietnam/ceremony_country/partner_local/same_sex)
[GOV.UK](https://gov.uk/marriage-abroad/y/vietnam/ceremony_country/partner_local/same_sex)

## Expected changes

- Update content for same sex marrigages where partner is Vietnamese National

#### Before
screen shot 2016-09-13 at 10 22 03

#### After
screen shot 2016-09-13 at 10 22 10

## Factcheck Content and Links Update

[Preview link](https://smart-answers-pr-2721.herokuapp.com/marriage-abroad/y/vietnam/ceremony_country/partner_local/opposite_sex)
[GOV.UK](https://gov.uk/marriage-abroad/y/vietnam/ceremony_country/partner_local/opposite_sex)

## Expected changes

- Update Content and links for Opposite and same sex marriages in Vietnam

#### Before

screen shot 2016-09-13 at 10 23 54

#### After
screen shot 2016-09-13 at 10 24 19

### CODE REVIEW

File: lib/data/marriage_abroad_services.yml [hunk 1]
```diff
@@ -160,11 +160,7 @@ vietnam:
       - :affidavit_or_affirmation_for_marriage
   same_sex:
     default:
-      partner_other:
-      - :receiving_notice_of_registration
-      - :registering_civil_partnership_or_marriage
-      - :issuing_civil_partnership_or_marriage_certificate
-      partner_british:
+      default:
```
Comment (pmanrubia) [inline lib/data/marriage_abroad_services.yml]: Could you please create a commit with the services update and update the artefacts in the same commit?

File: lib/smart_answer_flows/marriage-abroad/outcomes/_appointment_documents_in_vietnam.govspeak.erb [hunk 2]
```diff
@@ -1,9 +1,20 @@
-- a [service request form](/government/publications/service-request-form)
-- a [notice of marriage form](/government/publications/notice-of-same-sex-marriage-vietnam)
-- an [affirmation](/government/publications/affirmation-of-marital-status) or [affidavit](/government/publications/affidavit-of-marital-status) of marital status
+<% if calculator.ceremony_country == 'vietnam' &&
```
Comment (pmanrubia) [inline lib/smart_answer_flows/marriage-abroad/outcomes/_appointment_documents_in_vietnam.govspeak.erb]: if the partial is named: `_appointment_documents_in_vietnam` why do we need this check?

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_same_sex_marriage_and_civil_partnership.govspeak.erb [hunk 3]
```diff
@@ -34,12 +34,14 @@
...
   <% elsif calculator.ceremony_country == 'vietnam' %>
-    You’ll need to download and complete the following documents:
-
-    <%= render partial: "appointment_documents_in_vietnam.govspeak.erb" %>
+    <% unless calculator.partner_is_national_of_ceremony_country?  %>
+      You’ll need to download and complete the following documents:
+    <% end %>
 
-    The embassy will confirm if your marriage can be registered and give you a link to book a ‘notice of marriage’ appointment online and a reference number. You must quote the reference number when booking your appointment.
+    <%= render partial: "appointment_documents_in_vietnam.govspeak.erb",
```
Comment (leenagupte) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_same_sex_marriage_and_civil_partnership.govspeak.erb]: @ikennaokpala Shouldn't the partial `appointment_documents_in_vietnam.govspeak.erb` be added to the `unless` statement?

File: lib/data/marriage_abroad_services.yml (hunk 1 above)
Comment (leenagupte) [inline lib/data/marriage_abroad_services.yml]: @ikennaokpala I don't understand what this change does. Can you split it into it's commit, do I can see what gets updated?

File: lib/smart_answer/calculators/marriage_abroad_data_query.rb [hunk 4]
```diff
@@ -89,7 +89,7 @@ def ss_alt_fees_table_country?(country_slug, calculator)
 
     def ss_marriage_not_possible?(country_slug, calculator)
       (SS_ALT_FEES_TABLE_OR_OUTCOME_GROUP_A.include?(country_slug) && calculator.partner_not_british?) ||
-        ((SS_ALT_FEES_TABLE_OR_OUTCOME_GROUP_B.include?(country_slug) || %w(cambodia vietnam).include?(country_slug)) && calculator.partner_is_national_of_ceremony_country?) ||
+        ((SS_ALT_FEES_TABLE_OR_OUTCOME_GROUP_B.include?(country_slug) || %w(cambodia).include?(country_slug)) && calculator.partner_is_national_of_ceremony_country?) ||
```
Comment (leenagupte) [inline lib/smart_answer/calculators/marriage_abroad_data_query.rb]: @ikennaokpala What happens when Vietnam is removed from this check? Are any test artefacts affected?

**Comment by leenagupte** [comment]
@ikennaokpala There is a lot going on in commit bae1324 that makes it difficult to review.

I would move changes related to `_appointment_documents_in_vietnam.govspeak.erb` into a separate commit, and regenerate the test artefacts just for these changes.

I would also move the changes related to `_convert_cc_to_ss_marriage.govspeak.erb` into a separate commit, and regenerate the test artefacts just for these changes.

**Comment by leenagupte** [comment]
@ikennaokpala You need to change the description of commit c1c13c4 because, it's not just the opposite sex outcomes that are being changed, and preferably, split the change to add a callout to same-sex outcomes (where the partner is Vietnamese local) to a separate commit.

File: lib/data/marriage_abroad_services.yml (hunk 1 above)
Comment (ikennaokpala) [inline lib/data/marriage_abroad_services.yml]: Noted.. will make this it own commit

File: lib/smart_answer/calculators/marriage_abroad_data_query.rb (hunk 4 above)
Comment (ikennaokpala) [inline lib/smart_answer/calculators/marriage_abroad_data_query.rb]: well this affects only same sex marriages where the partner is vietnamese.. and yes it does produce artefacts.. I will demotrate that in its own commit..

File: lib/smart_answer_flows/marriage-abroad/outcomes/_appointment_documents_in_vietnam.govspeak.erb (hunk 2 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad/outcomes/_appointment_documents_in_vietnam.govspeak.erb]: Thanks pablo

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_same_sex_marriage_and_civil_partnership.govspeak.erb (hunk 3 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_same_sex_marriage_and_civil_partnership.govspeak.erb]: at the time of making changes this did not really occur to be.. I will try it out and see what comes out from the artefacts..

**Comment by ikennaokpala** [comment]
@pmanrubia and @leenagupte I hope the splits clarify things

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_same_sex_marriage_and_civil_partnership.govspeak.erb [hunk 5]
```diff
@@ -34,7 +34,9 @@
                locals: { calculator: calculator } %>
 
   <% elsif calculator.ceremony_country == 'vietnam' %>
-    You’ll need to download and complete the following documents:
+    <% unless calculator.partner_is_national_of_ceremony_country?  %>
+      You’ll need to download and complete the following documents:
+    <% end %>
```
Comment (leenagupte) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_same_sex_marriage_and_civil_partnership.govspeak.erb]: @ikennaokpala As this text only ever refers to Vietnam's appointment documents, I would move it to partial `_appointment_documents_in_vietnam.govspeak.erb`

**Comment by leenagupte** [comment]
@ikennaokpala This much better! Thanks!

Just a few things:

1) There are only 3 changed test artefacts, so I would squash commit 09bb5e29d976d5f3571065c71db4389bc1fdee61 into 23f054ca7fa7dc512d525c2cacf8790d2de394ab

2) Can you update the commit note to say that previously, people who weren't Vietnamese nationals could have a same-sex marriage in Vietnam, and that this has now been expanded to allow Vietnamese nationals to marry as well. This is why no further changes were needed to the generic same sex outcome to account for Vietnam.

3) There are only 3 changed test artefacts, so I would squash commit dea2a8e149d3855a6d4e5fdf82620ef9dbc9500f into 156a071503c1f6ef58f951adea509ded840eea13

4) There are only 3 changed test artefacts, so I would squash commit 7c80020434f10d8ef5a5294b9c76987579be14fa into af7e9ff0d8ac2ac3b48e1c3a87e8c8a4b387ee48

5) In commit 2a7372661d4c5eb45ee740ccfba726847150dce9, there is a condition around the text, but in the google doc attached, no condition exists. 
    
    a) If the condition is no longer required, delete this commit and commit 482dcb8d7652a6db2fb3eec18445c7702c4e0f57
    
    b) If the condition is required, as there are only 3 changed test artefacts, so I would squash commit 482dcb8d7652a6db2fb3eec18445c7702c4e0f57 into 2a7372661d4c5eb45ee740ccfba726847150dce9

6) There are only 3 changed test artefacts, so I would squash commit ce999ada38017453df4b93e3871b0deab3a1ad18 into 312cc96e2ab5b247e71310673ffd00ac3db9c93a

7) I think you should split commit 1f531df0d3a67093430ed8c99279e76ce9e26e36 into 1) the changes that affect same-sex marriages where the partner is a Vietnamese local, and 2) all other Vietnam outcomes.

**Comment by leenagupte** [comment]
@ikennaokpala This much better! Thanks!

Just a few things:

1) There are only 3 changed test artefacts, so I would squash commit 09bb5e29d976d5f3571065c71db4389bc1fdee61 into 23f054ca7fa7dc512d525c2cacf8790d2de394ab

2) Can you update the commit note to say that previously, people who weren't Vietnamese nationals could have a same-sex marriage in Vietnam, and that this has now been expanded to allow Vietnamese nationals to marry as well. This is why no further changes were needed to the generic same sex outcome to account for Vietnam.

3) There are only 3 changed test artefacts, so I would squash commit dea2a8e149d3855a6d4e5fdf82620ef9dbc9500f into 156a071503c1f6ef58f951adea509ded840eea13

4) There are only 3 changed test artefacts, so I would squash commit 7c80020434f10d8ef5a5294b9c76987579be14fa into af7e9ff0d8ac2ac3b48e1c3a87e8c8a4b387ee48

5) In commit 2a7372661d4c5eb45ee740ccfba726847150dce9, there is a condition around the text, but in the google doc attached, no condition exists.

a) If the condition is no longer required, delete this commit and commit 482dcb8d7652a6db2fb3eec18445c7702c4e0f57

b) If the condition is required, as there are only 3 changed test artefacts, so I would squash commit 482dcb8d7652a6db2fb3eec18445c7702c4e0f57 into 2a7372661d4c5eb45ee740ccfba726847150dce9
6) There are only 3 changed test artefacts, so I would squash commit ce999ada38017453df4b93e3871b0deab3a1ad18 into 312cc96e2ab5b247e71310673ffd00ac3db9c93a

7) I think you should split commit 1f531df0d3a67093430ed8c99279e76ce9e26e36 into 1) the changes that affect same-sex marriages where the partner is a Vietnamese local, and 2) all other Vietnam outcomes.

**Comment by ikennaokpala** [comment]
I am closing this PR, becuase after chating with @amylaurence. things have now changed and need to be addressed differently

**Comment by ikennaokpala** [comment]
I am closing this PR, becuase after chating with @amylaurence. things have now changed and need to be addressed differently
